fix: browse course material button styling

MAbdurrehman12 opened the pull request:
#### Pre-Flight checklist

- [x] Screenshots and design review for any changes that affect layout or styling
  - [x] Desktop screenshots
  - [x] Mobile width screenshots
- [x] Testing
  - [x] Code is tested
  - [x] Changes have been manually tested

#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-hugo-themes/issues/606

#### What's this PR do?
- Changes the style of `browse course material` button according to issue ticket and invision design
- Removes the borderline for medium and below screens making the button a separator, but on large or above screens, borderline will be shown.
- Adds some space around the button on course home page 

#### How should this be manually tested?
- Checkout this branch
- Build any course(s) locally
- Go to a smaller screen size (tablet or less)
- Verify that the style `browse course material` button is correct.

#### Screenshots (if appropriate)

<img width="865" alt="image" src="https://user-images.githubusercontent.com/93309234/161949604-7f73b56b-da9d-4c31-b1e2-a5123e921f0a.png">

<img width="873" alt="image" src="https://user-images.githubusercontent.com/93309234/161949567-e788e7b7-0021-4620-93e0-5945fecb5f14.png">

<img width="383" alt="image" src="https://user-images.githubusercontent.com/93309234/161949481-92d13f14-b450-44f6-bf34-0192b48f9b55.png">

<img width="384" alt="image" src="https://user-images.githubusercontent.com/93309234/161949380-0635fab6-af90-4bcb-a448-2c27172a3279.png">

<img width="383" alt="image" src="https://user-images.githubusercontent.com/93309234/161949354-0b370821-659c-4445-957c-1ccf85ebede0.png">

abeglova commented:
This works but it looks like the mobile design no longer has a top border line above the button. I think that looks better than keeping that border

abeglova reviewed (changes requested):
One comment about the border line. Also, there is a lot of variation in how much whitespace there is around the "browse course materials button"

MAbdurrehman12 commented:
> This works but it looks like the mobile design no longer has a top border line above the button. I think that looks better than keeping that border

The border is not with this `course material` button, it's with/below the heading section, and since it wasn't mentioned in the issue ticket so I did not remove it despite the fact it's not there in the design but I can remove it if you think it will look better without the border.  

> there is a lot of variation in how much whitespace there is around the "browse course materials button"

Not sure if I follow this. Variation in different screen sizes?

pdpinch commented:
> The border is not with this course material button, it's with/below the heading section, and since it wasn't mentioned in the issue ticket so I did not remove it despite the fact it's not there in the design but I can remove it if you think it will look better without the border.

If it's not too much trouble, please remove it. The button is enough of a visual separator. We don't need a line as well.

abeglova commented:
The variation is by page type: The button has less white space around it on course home pages than resource pages

MAbdurrehman12 commented:
@abeglova Thanks for reviewing and pointing out these things. 
I have removed the borderline on resources pages and added some space around the button in course home page.
Can you please have another look?

btw, I think there is an inconsistency in right/left spacing in course home page and resources pages. (course home page has more space on the left and right sides as compared to resource pages)

e.g:
course home page: 
<img width="85" alt="image" src="https://user-images.githubusercontent.com/93309234/161951682-d43d1dc7-3202-41db-b81b-1e335a00d13f.png">

resource page:
<img width="94" alt="image" src="https://user-images.githubusercontent.com/93309234/161951503-79e9e0cb-2bc5-4962-a3ab-4be00f6570ab.png">

not sure if this is something that needs to be catered too.

abeglova reviewed (approved):
LGTM
